easybuild/easyblocks/c/cp2k.py

```diff
@@ -231,9 +232,11 @@ def configure_step(self):
         # CUDA
         cuda = get_software_root('CUDA')
         if cuda:
-            options['DFLAGS'] += ' -D__ACC -D__DBCSR_ACC'
+            options['DFLAGS'] += ' -D__ACC -D__DBCSR_ACC -D__PW_CUDA'
             options['LIBS'] += ' -lcudart -lcublas -lcufft -lrt'
             options['NVCC'] = ' nvcc'
+            self.log.info("Using Nvidia arch: %s" % self.cfg['nvarch'])
+            options['NVFLAGS'] = options['DFLAGS'] + ' -O3 -arch ' + self.cfg['nvarch']
```

Comment:
@Micket This should be under an `if self.cfg['nvarch']`, and we should avoid hardcoding the `-O3`?

Reply:
I put in a -O3 because this was what other CP2K example build scripts did. Appearantly, this is just the build flags for the CPU code (the gpu code has the equivalent O3 level by default) **correct me if I'm wrong**. But, since this is the CPU code, perhaps we should put a value here based on the toolchainopts?

Reply:
Indeed, this should ideally be based on what the toolchain sets.

Not sure how trivial it is to only get the `-O2` (or whatever) from `self.toolchain` though...

Reply:
@boegel I'd rather expose extra_nvflags and let the config decide?

Reply:
I've done it slightly different:

```diff
@@ -103,6 +103,7 @@ class EB_CP2K(EasyBlock):
             'plumed': [None, "Enable PLUMED support", CUSTOM],
             'type': ['popt', "Type of build ('popt' or 'psmp')", CUSTOM],
             'typeopt': [True, "Enable optimization", CUSTOM],
+            'cuda_cc': [[], "List of CUDA compute capability to compile for", CUSTOM],
         }
         return EasyBlock.extra_options(extra_vars)
 
@@ -231,9 +232,14 @@ class EB_CP2K(EasyBlock):
         # CUDA
         cuda = get_software_root('CUDA')
         if cuda:
-            options['DFLAGS'] += ' -D__ACC -D__DBCSR_ACC'
+            options['DFLAGS'] += ' -D__ACC -D__DBCSR_ACC -D__PW_CUDA'
             options['LIBS'] += ' -lcudart -lcublas -lcufft -lrt'
             options['NVCC'] = ' nvcc'
+            self.log.info("Using CUDA Compute capabilities: %s" % self.cfg['cuda_cc'])
+            capabilities = [''.join(x.split('.')) for x in self.cfg['cuda_cc']]
+            options['NVFLAGS'] = options['DFLAGS'] + ' '
+            options['NVFLAGS'] += ' '.join(['-gencode arch=compute_%s,code=\"sm_%s,compute_%s\"' % (x, x, x)
+                                            for x in capabilities])
 
```

To we can set in the easyconfig:
```python
cuda_cc = ['3.5', '6.0']
```

Reply:
@wpoely86 I was about to add a similar change, but I was holding off to deal with the rest of the compile flags for nvcc. I'm not sure if they matter greatly, but CP2K's example build files suggested the use of -O3 at least.

Reply:
These makefiles are just examples. If you want to use `-O3`, that's fine but make it an option.